src/Databases/DataLake/GlueCatalog.cpp

```diff
@@ -0,0 +1,417 @@
+        auto args = splitTypeArguments(inner);
+
+        std::vector<String> field_names;
+        std::vector<DB::DataTypePtr> field_types;
+
+        for (const auto & arg : args)
+        {
+            size_t colon = arg.find(':');
+            if (colon == String::npos)
+                throw DB::Exception(DB::ErrorCodes::DATALAKE_DATABASE_ERROR, "Invalid data type {}", type_name);
+
+            String field_name = trim(arg.substr(0, colon));
+            String field_type = trim(arg.substr(colon + 1));
+            String full_field_name = prefix.empty() ? field_name : prefix + "." + field_name;
+
+            field_names.push_back(full_field_name);
+            field_types.push_back(getType(field_type, nullable, full_field_name));
+        }
+        return std::make_shared<DB::DataTypeTuple>(field_types, field_names);
+    }
+
+    return nullable ? DB::makeNullable(DB::IcebergSchemaProcessor::getSimpleType(name)) : DB::IcebergSchemaProcessor::getSimpleType(name);
+}
+
+}
+
+namespace DataLake
+{
+
+GlueCatalog::GlueCatalog(
+    const String & access_key_id,
+    const String & secret_access_key,
+    const String & region_,
+    const String & endpoint,
+    DB::ContextPtr context_)
+    : ICatalog("")
+    , DB::WithContext(context_)
+    , log(getLogger("GlueCatalog(" + region_ + ")"))
+    , credentials(access_key_id, secret_access_key)
+    , region(region_)
+{
+    DB::S3::CredentialsConfiguration creds_config;
+    creds_config.use_environment_credentials = true;
+
+    const DB::Settings & global_settings = getContext()->getGlobalContext()->getSettingsRef();
+
+    int s3_max_redirects = static_cast<int>(global_settings[DB::Setting::s3_max_redirects]);
+    int s3_retry_attempts = static_cast<int>(global_settings[DB::Setting::s3_retry_attempts]);
+    bool enable_s3_requests_logging = global_settings[DB::Setting::enable_s3_requests_logging];
+    DB::S3::PocoHTTPClientConfiguration poco_config = DB::S3::ClientFactory::instance().createClientConfiguration(
+        region,
+        getContext()->getRemoteHostFilter(),
+        s3_max_redirects,
+        s3_retry_attempts,
+        enable_s3_requests_logging,
+        false,
+        nullptr,
+        nullptr
+    );
+
+    Aws::Glue::GlueClientConfiguration client_configuration;
+    client_configuration.maxConnections = static_cast<unsigned>(global_settings[DB::Setting::s3_max_connections]);
+    client_configuration.connectTimeoutMs = 10 * 1000;
+    client_configuration.requestTimeoutMs = 30 * 1000;
+    client_configuration.region = region;
+    auto endpoint_provider = std::make_shared<Aws::Glue::GlueEndpointProvider>();
+
+    /// Only for testing when we are mocking glue
+    if (!endpoint.empty())
+    {
+        client_configuration.endpointOverride = endpoint;
+        endpoint_provider->OverrideEndpoint(endpoint);
+        Aws::Auth::AWSCredentials fake_credentials_for_fake_catalog;
+        if (credentials.IsEmpty())
+        {
```

Comment:
Should we do LOG_WARNING here?

Reply:
No, because there are 500 different ways how we can authenticate in AWS without explicit credentials.